API Documentation

```diff
@@ -0,0 +1,740 @@
+# Chiquito API Documentation
+
+# Overall Workflow
+
+Chiquito is a DSL that compiles Chiquito AST to an IR which can be parsed by a Chiquito Halo2 backend and integrated into a Halo2 circuit. Therefore, to create a Halo2 circuit using Chiquito, we need to:
+
+- Call `circuit` function in DSL, which returns an `ast::Circuit` object. The `circuit` function defines the `ast::Circuit` object by:
+    - Importing Halo2 columns
+    - Generating fixed columns
+    - Adding forward signals
+    - Defining and instantiating “steps”, which defines internal signals, constraints, lookups, and witness generation
+- Create a `Compiler` object that compiles the `ast::Circuit` object to an `ir::Circuit` object
+- Call `chiquito2Halo2` function in Halo2 backend to convert the `ir::Circuit` object to a `ChiquitoHalo2` object
+- Integrate `ChiquitoHalo2` object into a Halo2 circuit using a boiler plate code
+
+# DSL Functions
+
+The above section describes the high level process of building and integrating a Chiquito Halo2 backend object into a Halo2 circuit. However, when building a circuit using Chiquito, the developer mostly call DSL functions to manipulate the `ast::Circuit` object, which the rest of this API documentation will focus on.
+
+Before we dive in, there are a few Chiquito-specific concepts that are not in Halo2:
+
+- Step (or super row): Chiquito provides syntax sugar for defining complex constraints and therefore sometimes requires allocating multiple Halo2 rows to build out a set of constraints. These groups of rows are called “super rows” or “steps”.
```

Comment:
a super row is the rows and columns that a step instance uses in the plonkish table

Reply:
Edits:
- Step (or super row): Each "step" uses one or multiple rows and columns in the plonkish table to define constraints and witness generation. That's why "steps" are also called "super rows".

Reply:
A super row is only related to the step instance, not the the step definition. A step definition has no rows, jury constraints

Reply:
fixed